File: Java Barcode Scanner ME/app/src/main/java/de/dimedis/mobileentry/fragments/menus/DeviceSettingsMenuFragment.java
package de.dimedis.mobileentry.fragments.menus;


import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.SeekBar;

import androidx.annotation.NonNull;

import de.dimedis.mobileentry.databinding.FragmentMenuDeviceSettingsBinding;

public class DeviceSettingsMenuFragment extends AbsSettingsMenuFragment {
    private static final String TAG = "Device Settings";

    FragmentMenuDeviceSettingsBinding binding;

    public static Builder builder() {
        return new Builder();
    }

    public static class Builder {
        public Builder() {
        }

        public DeviceSettingsMenuFragment build() {
            return new DeviceSettingsMenuFragment();
        }
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        super.onCreateView(inflater, container, savedInstanceState);

        binding = FragmentMenuDeviceSettingsBinding.inflate(inflater, container, false);
        binding.okCancel.ok.setOnClickListener(view -> onClickOK());
        binding.okCancel.cancel.setOnClickListener(view -> onClickCancel());

        return binding.getRoot();
    }

    @Override
    protected void init() {
        super.init();
        binding.vibration.setChecked(mSettings.isVibration());
        binding.cameraReaderSetting.setChecked(mSettings.isCamReader());
        binding.sound.setMax(mSettings.getSoundMax());
        binding.sound.setProgress(mSettings.getSound());
        binding.brightness.setMax(mSettings.getBrightnessMax());
        binding.brightness.setProgress(mSettings.getBrightness());
        binding.permanentScanModeSetting.setChecked(mSettings.isPermScanMode());
        binding.vibration.setOnCheckedChangeListener((buttonView, isChecked) -> mSettings.setVibration(isChecked));
        binding.cameraReaderSetting.setOnCheckedChangeListener((buttonView, isChecked) -> mSettings.setCamReader(isChecked));
        binding.permanentScanModeSetting.setOnCheckedChangeListener((buttonView, isChecked) -> mSettings.setPermScanMode(isChecked));
        binding.sound.setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {

            @Override
            public void onProgressChanged(SeekBar seekBar, int progress,
                                          boolean fromUser) {
                if (fromUser) {
                    mSettings.setSound(progress);
                    seekBar.setProgress(progress);
                }
            }

            @Override
            public void onStartTrackingTouch(SeekBar seekBar) {
            }

            @Override
            public void onStopTrackingTouch(SeekBar seekBar) {
            }
        });

        binding.brightness.setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {
            @Override
            public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
                mSettings.setBrightness(progress);
            }

            @Override
            public void onStartTrackingTouch(SeekBar seekBar) {
            }

            @Override
            public void onStopTrackingTouch(SeekBar seekBar) {
            }
        });
    }
}

File: Kotlin Migration/NTMobileEntry/app/src/main/java/de/dimedis/mobileentry/fragments/menus/DeviceSettingsMenuFragment.kt
package de.dimedis.mobileentry.fragments.menus

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.CompoundButton
import android.widget.SeekBar.OnSeekBarChangeListener
import android.widget.SeekBar
import de.dimedis.mobileentry.databinding.FragmentMenuDeviceSettingsBinding

class DeviceSettingsMenuFragment : AbsSettingsMenuFragment() {
    var binding: FragmentMenuDeviceSettingsBinding? = null

    class Builder {
        fun build(): DeviceSettingsMenuFragment {
            return DeviceSettingsMenuFragment()
        }
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        super.onCreateView(inflater, container, savedInstanceState)
        binding = FragmentMenuDeviceSettingsBinding.inflate(inflater, container, false)
        binding!!.okCancel.ok.setOnClickListener {
            onClickOK() }
        binding!!.okCancel.cancel.setOnClickListener {
            onClickCancel() }
        return binding!!.root
    }

    override fun init() {
        super.init()
        binding!!.vibration.isChecked = mSettings!!.vibration
        binding!!.cameraReaderSetting.isChecked = mSettings!!.camReader
        binding!!.sound.max = mSettings!!.soundMax
        binding!!.sound.progress = mSettings!!.sound
        binding!!.brightness.max = mSettings!!.brightnessMax
        binding!!.brightness.progress = mSettings!!.brightness
        binding!!.permanentScanModeSetting.isChecked = mSettings!!.permScanMode
        binding!!.vibration.setOnCheckedChangeListener { buttonView: CompoundButton?, isChecked: Boolean ->
            mSettings!!.vibration = isChecked
        }
        binding!!.cameraReaderSetting.setOnCheckedChangeListener { buttonView: CompoundButton?, isChecked: Boolean ->
            mSettings!!.camReader = isChecked
        }
        binding!!.permanentScanModeSetting.setOnCheckedChangeListener { buttonView: CompoundButton?, isChecked: Boolean ->
            mSettings!!.permScanMode = isChecked
        }
        binding!!.sound.setOnSeekBarChangeListener(object : OnSeekBarChangeListener {

        override fun onProgressChanged(seekBar: SeekBar, progress: Int, fromUser: Boolean) {
                if (fromUser) {
                    mSettings!!.sound = progress
                    seekBar.progress = progress
                }
            }
        override fun onStartTrackingTouch(seekBar: SeekBar?) {}
        override fun onStopTrackingTouch(seekBar: SeekBar?) {}
        })

        binding!!.brightness.setOnSeekBarChangeListener(object : OnSeekBarChangeListener {
            override fun onProgressChanged(seekBar: SeekBar?, progress: Int, fromUser: Boolean) {
                mSettings!!.brightness = progress
            }

            override fun onStartTrackingTouch(seekBar: SeekBar?) {}
            override fun onStopTrackingTouch(seekBar: SeekBar?) {}
        })
    }

    companion object {
        private const val TAG = "Device Settings"
        fun builder(): Builder {
            return Builder()
        }
    }
}